cvmfs/receiver/commit_processor.cc

```diff
@@ -100,13 +102,24 @@ CommitProcessor::Result CommitProcessor::Process(
   RepositoryTag final_tag = tag;
   // If tag_name is a generic tag, update the time stamp
   if (HasPrefix(final_tag.name_, "generic-", false)) {
-    // timestamp=$(date -u "+%Y-%m-%dT%H:%M:%SZ")
-    char buf[32];
-    time_t now = time(NULL);
+    // format time following the ISO 8601 YYYY-MM-DDThh:mm:ss.sssZ
+    // note the millisecond accurracy
+    struct timespec now;
     struct tm timestamp;
-    gmtime_r(&now, &timestamp);
-    strftime(buf, sizeof(buf), "%Y-%m-%dT%H:%M:%SZ", &timestamp);
-    final_tag.name_ = std::string("generic-") + buf;
+    clock_gettime(CLOCK_REALTIME, &now);
```

Comment:
We should make this a posix.cc utility function; we also need a `platform_...` function because `get_clock` is Linux only